libasi/CMakeLists.txt

```diff
@@ -0,0 +1,133 @@
+    add_definitions(-DLOW_USB_BANDWIDTH)
+  ELSEIF (CMAKE_SYSTEM_PROCESSOR MATCHES "armv7")
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/armv7/libASICamera2.bin ${CMAKE_BINARY_DIR}/libASICamera2.so.${ASICAM_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/armv7/libEFWFilter.bin ${CMAKE_BINARY_DIR}/libEFWFilter.so.${ASIEFW_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/armv7/libUSB2ST4Conv.bin ${CMAKE_BINARY_DIR}/libUSB2ST4Conv.so.${ASIST4_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/armv7/libEAFFocuser.bin ${CMAKE_BINARY_DIR}/libEAFFocuser.so.${ASIEAF_VERSION})
+    add_definitions(-DLOW_USB_BANDWIDTH)
+  ELSEIF (CMAKE_SYSTEM_PROCESSOR MATCHES "armv8")
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/armv8/libASICamera2.bin ${CMAKE_BINARY_DIR}/libASICamera2.so.${ASICAM_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/armv8/libEFWFilter.bin ${CMAKE_BINARY_DIR}/libEFWFilter.so.${ASIEFW_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/armv8/libUSB2ST4Conv.bin ${CMAKE_BINARY_DIR}/libUSB2ST4Conv.so.${ASIST4_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/armv8/libEAFFocuser.bin ${CMAKE_BINARY_DIR}/libEAFFocuser.so.${ASIEAF_VERSION})
+    add_definitions(-DLOW_USB_BANDWIDTH)
+  ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/x64/libASICamera2.bin ${CMAKE_BINARY_DIR}/libASICamera2.so.${ASICAM_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/x64/libEFWFilter.bin ${CMAKE_BINARY_DIR}/libEFWFilter.so.${ASIEFW_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/x64/libUSB2ST4Conv.bin ${CMAKE_BINARY_DIR}/libUSB2ST4Conv.so.${ASIST4_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/x64/libEAFFocuser.bin ${CMAKE_BINARY_DIR}/libEAFFocuser.so.${ASIEAF_VERSION})
+  ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "i[3-6]86")
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/x86/libASICamera2.bin ${CMAKE_BINARY_DIR}/libASICamera2.so.${ASICAM_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/x86/libEFWFilter.bin ${CMAKE_BINARY_DIR}/libEFWFilter.so.${ASIEFW_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/x86/libUSB2ST4Conv.bin ${CMAKE_BINARY_DIR}/libUSB2ST4Conv.so.${ASIST4_VERSION})
+    exec_program(cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/x86/libEAFFocuser.bin ${CMAKE_BINARY_DIR}/libEAFFocuser.so.${ASIEAF_VERSION})
+  ENDIF ()
+ENDIF(UNIX AND NOT WIN32)
+
+install( FILES ${CMAKE_BINARY_DIR}/libASICamera2.so.${ASICAM_VERSION} DESTINATION ${LIB_INSTALL_DIR}${LIB_POSTFIX})
+install( FILES ${CMAKE_BINARY_DIR}/libEFWFilter.so.${ASIEFW_VERSION} DESTINATION ${LIB_INSTALL_DIR}${LIB_POSTFIX})
+install( FILES ${CMAKE_BINARY_DIR}/libUSB2ST4Conv.so.${ASIST4_VERSION} DESTINATION ${LIB_INSTALL_DIR}${LIB_POSTFIX})
+install( FILES ${CMAKE_BINARY_DIR}/libEAFFocuser.so.${ASIEAF_VERSION} DESTINATION ${LIB_INSTALL_DIR}${LIB_POSTFIX})
```

Comment:
One very important thing that I forgot is that all these files are archive and not shared libraries! So how did you link with them?

Reply:
Oh, that's interesting. They link without problem during compilation time and the binary works ok. I have tested it on my ASI camera before PR

Reply:
I have just run ldconfig and it returns "is not an ELF file - it has the wrong magic bytes at the start" for each of these files. Also file returns "current ar archive". What can we do about it?

Reply:
I believe ASI releases both .ar and .so versions.. can you try their latest SDK and see if it resolves this issue?

Reply:
I got SDKs and they provide so files indeed. The problem is that the SDK for ST4 improperly linked on ZWO site so I'm missing this library (USB2ST4_linux_mac_SDK)

Reply:
This product is discontinued (see https://astronomy-imaging-camera.com/product/zwo-usbst4-guider-adapter) so they don't provide SDK. If you have the SDK for it I can include it, otherwise this lib should be linked the old way in indi-asi :-/

Reply:
Ok, so maybe stick to static linking for that component?

Reply:
I believe I have a copy of the SDK including the shared objects, if that would be helpful?

Reply:
James, that would be just perfect. Could you drop me an email with it?

Reply:
I have everything but USB2ST4_linux_mac_SDK

Reply:
Should be on its way.

Reply:
All done. Thanks for the libs James!